File: docs/snippet/code-signal/arcade/intro/edge-of-the-ocean/matrix-elements-sum/matrix-elements-sum.py
def matrix_elements_sum(matrix):
    rent = 0
    for section in range(len(matrix[0])):
        for room in range(len(matrix)):
            if matrix[room][section] == 0:
                # ghost found, exit the entire section below
                break
            rent += matrix[room][section]
    return rent


print(matrix_elements_sum([
    [0, 1, 1, 2],
    [0, 5, 0, 0],
    [2, 0, 3, 3]
]))

File: content/snippet/code-signal/arcade/intro/edge-of-the-ocean/matrix-elements-sum/matrix-elements-sum.js
function matrixElementsSum(matrix) {
  let rent = 0;
  for (let section = 0; section < matrix[0].length; section++) {
    for (let room = 0; room < matrix.length; room++) {
      if (matrix[room][section] === 0) {
        break;
      }
      rent += matrix[room][section];
    }
  }
  return rent;
}

console.log(
  matrixElementsSum([
    [0, 1, 1, 2],
    [0, 5, 0, 0],
    [2, 0, 3, 3],
  ])
);
